src/test/java/com/ericsson/ei/utils/MongoUriTest.java

```diff
@@ -0,0 +1,70 @@
+/*
+   Copyright 2019 Ericsson AB.
+   For a full list of individual contributors, please see the commit history.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package com.ericsson.ei.utils;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+
+import com.ericsson.ei.mongo.MongoUri;
+
+public class MongoUriTest {
+
+    private static final String USERNAME = "username";
+    private static final String PASSWORD = "my_password";
```

Comment:
Connected to @saif-ericsson comment above. Where do we handle passwords like `P@sswo:d`

Reply:
Will investigate this, if that should be allowed (uri works with such passwords) or we need to improve password extraction algorithm.

Reply:
Passwordextraction is now handled using com.mongodb.MongoClientURI.

: and @ in passwords muct be url encoded and exception will be thrown.